common/persistence/visibility/store/sql/visibility_store.go

```diff
@@ -614,14 +739,18 @@ func (s *VisibilityStore) rowToInfo(
 		RootWorkflowID: row.RootWorkflowID,
 		RootRunID:      row.RootRunID,
 		Memo:           persistence.NewDataBlob(row.Memo, row.Encoding),
-		ChasmMemo:      persistence.NewDataBlob(row.ChasmMemo, row.Encoding),
 	}
 	if row.SearchAttributes != nil && len(*row.SearchAttributes) > 0 {
-		searchAttributes, err := s.processRowSearchAttributes(*row.SearchAttributes, nsName, chasmMapper)
+		// Always separate CHASM and custom search attributes
+		chasmSAs, customSAs, err := s.processRowSearchAttributesSeparated(*row.SearchAttributes, nsName, chasmMapper)
```

Comment:
We should split and alias the search attributes in the visibility manager instead of visibility store. Another PR to refactor this later.